<!-- -->
<launch>

  <arg name="initialize_source" default="Rviz"/>
  <arg name="loop_rate" default="100.0"/>
  <arg name="angle_error" default="0.0"/>
  <arg name="position_error" default="0.0"/>
  <arg name="lidar_height" default="1.0"/>
  <arg name="use_ctrl_cmd" default="false" />
  <arg name="sim_pose_name" default="sim_pose" />
  <arg name="sim_velocity_name" default="sim_velocity" />
  <arg name="sim_vehicle_status_name" default="vehicle_status" />
  <arg name="simulation_frame_id" default="sim_base_link" />
  <arg name="lidar_frame_id" default="sim_lidar" />
  <arg name="map_frame_id" default="map" />
  

  <!-- vehicle model parameters -->
  <arg name="vehicle_model_type" default="IDEAL_TWIST" />
  <arg name="tread_length" default="1.0" />
  <arg name="angvel_lim" default="3.0" />
  <arg name="vel_lim" default="10.0" />
  <arg name="steer_lim" default="1.0" />
  <arg name="accel_rate" default="1.0" />
  <arg name="angvel_rate" default="1.0" />
  <arg name="steer_vel" default="0.3" />
  <arg name="vel_time_delay" default="0.3" />
  <arg name="vel_time_constant" default="0.1" />
  <arg name="steer_time_delay" default="0.5" />
  <arg name="steer_time_constant" default="0.5" />
  <arg name="angvel_time_delay" default="0.2" />
  <arg name="angvel_time_constant" default="0.5" />

  
  <node pkg="waypoint_follower" type="wf_simulator" name="wf_simulator" output="screen">
    <param name="initialize_source" value="$(arg initialize_source)"/>
    <param name="loop_rate" value="$(arg loop_rate)"/>
    <param name="accel_rate" value="$(arg accel_rate)"/>
    <param name="angle_error" value="$(arg angle_error)"/>
    <param name="position_error" value="$(arg position_error)"/>
    <param name="lidar_height" value="$(arg lidar_height)"/>
    <param name="use_ctrl_cmd" value="$(arg use_ctrl_cmd)"/>

    <param name="sim_pose_name" value="$(arg sim_pose_name)" />
    <param name="sim_velocity_name" value="$(arg sim_velocity_name)" />
    <param name="sim_vehicle_status_name" value="$(arg sim_vehicle_status_name)" />
    <param name="simulation_frame_id" value="$(arg simulation_frame_id)" />
    <param name="lidar_frame_id" value="$(arg lidar_frame_id)" />
    <param name="map_frame_id" value="$(arg map_frame_id)" />

    <!-- vehicle model parameters -->
    <param name="vehicle_model_type" value="$(arg vehicle_model_type)" />
    <param name="tread_length" value="$(arg tread_length)" />
    <param name="angvel_lim" value="$(arg angvel_lim)" />
    <param name="vel_lim" value="$(arg vel_lim)" />
    <param name="steer_lim" value="$(arg steer_lim)" />
    <param name="accel_rate" value="$(arg accel_rate)" />
    <param name="angvel_rate" value="$(arg angvel_rate)" />
    <param name="steer_vel" value="$(arg steer_vel)" />
    <param name="vel_time_delay" value="$(arg vel_time_delay)" />
    <param name="vel_time_constant" value="$(arg vel_time_constant)" />
    <param name="steer_time_delay" value="$(arg steer_time_delay)" />
    <param name="steer_time_constant" value="$(arg steer_time_constant)" />
    <param name="angvel_time_delay" value="$(arg angvel_time_delay)" />
    <param name="angvel_time_constant" value="$(arg angvel_time_constant)" />


  </node>

</launch>
